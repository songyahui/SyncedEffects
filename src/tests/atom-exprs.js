"use strict"

var hh = require("hiphop");

var prg =
    <hh.module>
      <hh.loop>
	<hh.local L>
	  <hh.emit L value=${"foo bar"}/>
	  <hh.pause/>
	  <hh.atom apply=${function() {
	     console.log("atom works! value of L is", this.L.nowval);
	  }}/>
	</hh.local>
      </hh.loop>
    </hh.module>

exports.prg = new hh.ReactiveMachine(prg, "atom");
