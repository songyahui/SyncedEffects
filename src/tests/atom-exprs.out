atom> ;
--- Output:
atom> ;
atom works! value of L is foo bar
--- Output:
atom> ;
atom works! value of L is foo bar
--- Output:
atom> ;
atom works! value of L is foo bar
--- Output:
atom> ;
atom works! value of L is foo bar
--- Output:
atom> ;
atom works! value of L is foo bar
--- Output:
atom> ;
atom works! value of L is foo bar
--- Output:
atom> ;
atom works! value of L is foo bar
--- Output:
atom> ;
atom works! value of L is foo bar
--- Output:
atom> ;
atom works! value of L is foo bar
--- Output:
